src/bmp.cpp

```diff
@@ -30,20 +30,56 @@ const static BMPColourHeader colour_header = {
     0x000000ff,
     0xff000000,
     0x73524742,
-    {}};
+    {}
+};
 
-BMP::BMP(const char *filename)
+BMP::BMP(std::string filename)
 {
     read(filename);
-    m_background_value = get_average_colour();
-    m_non_background_count = get_non_background_pixel_count(m_background_value);
-    m_blurred_edge_mask = blur_edge_mask(sobel_edges<245>());
-    m_vertical_edges = filter_long_vertical_edge_runs(get_vertical_edges<245>(), 10);
+
+    m_blurred_edge_mask = blur_edge_mask();
+    m_filtered_edge_mask = filter_long_vertical_edge_runs(10);
 }
 
-BMP::BMP() {}
+BMP::BMP()
```

Comment:
this constructor is still problematic, as it at last leaves the `m_*_header` data members uninitialized, so that e.g. a following `get_width` would return garbage.  (And the default initialization of the other data members might or might not happen to fulfill the class's invariants---hard to tell for an instance for which even width and height are unspecified :)

The easiest thing might be to allow `width` and `height` for the following constructor to be zero (or would that cause any issues), and then let this constructor forward to that constructor,
```
BMP::BMP(): BMP(0, 0, false) {}
```